packages/airnode-node/src/evm/templates/template-fetching.test.ts

```diff
@@ -18,6 +18,74 @@ describe('fetch (templates)', () => {
     };
   });
 
+  it('uses valid templates from config without fetching from the chain', async () => {
+    mutableFetchOptions = {
+      ...mutableFetchOptions,
+      configTemplates: {
+        '0x38ba0e80224f14d0c654c4ba6e3745fcb7f310fd4f2f80994fe802da013edafe': {
+          endpointId: '0x13dea3311fe0d6b84f4daeab831befbc49e19e6494c41e9e065a09c3c68f43b6',
+          encodedParameters: '0x6874656d706c6174656576616c7565',
+        },
+        // invalid tempplateId
+        '0x38ba0e80224f14d0c654c4ba6e3745fcb7f310fd4f2f80994fe802da013edaxx': {
+          endpointId: '0x13dea3311fe0d6b84f4daeab831befbc49e19e6494c41e9e065a09c3c68f43b6',
+          encodedParameters: '0x6874656d706c6174656576616c7565',
+        },
+      },
+    };
+
+    const rawTemplates = {
+      airnodes: ['airnode-0', '0xD5659F26A72A8D718d1955C42B3AE418edB001e0'],
+      endpointIds: ['endpointId-0', '0x13dea3311fe0d6b84f4daeab831befbc49e19e6494c41e9e065a09c3c68f43b6'],
+      parameters: ['0x6874656d706c6174656576616c7565', '0x6874656d706c6174656576616c7565'],
+    };
+    getTemplatesMock.mockResolvedValueOnce(rawTemplates);
+
+    const apiCall1 = fixtures.requests.buildApiCall({
+      templateId: '0x38ba0e80224f14d0c654c4ba6e3745fcb7f310fd4f2f80994fe802da013edafe',
+    });
+    const apiCall2 = fixtures.requests.buildApiCall({ templateId: 'templateId-0' });
+    const apiCall3 = fixtures.requests.buildApiCall({
+      templateId: '0x38ba0e80224f14d0c654c4ba6e3745fcb7f310fd4f2f80994fe802da013edaxx',
+    });
+    const [logs, res] = await templates.fetch([apiCall1, apiCall2, apiCall3], mutableFetchOptions);
+    expect(logs).toEqual([]);
+    expect(res).toEqual({
+      'templateId-0': {
+        airnodeAddress: 'airnode-0',
+        endpointId: 'endpointId-0',
+        encodedParameters: '0x6874656d706c6174656576616c7565',
+        id: 'templateId-0',
+      },
+      // Invalid templateId
+      '0x38ba0e80224f14d0c654c4ba6e3745fcb7f310fd4f2f80994fe802da013edaxx': {
+        airnodeAddress: '0xD5659F26A72A8D718d1955C42B3AE418edB001e0',
+        endpointId: '0x13dea3311fe0d6b84f4daeab831befbc49e19e6494c41e9e065a09c3c68f43b6',
+        encodedParameters: '0x6874656d706c6174656576616c7565',
+        id: '0x38ba0e80224f14d0c654c4ba6e3745fcb7f310fd4f2f80994fe802da013edaxx',
+      },
```

Comment:
If we validate config templates then we might not even need to handle these invalid cases

Reply:
True, I'll add one to the validator instead